text/0000-try-expr.md

```diff
@@ -0,0 +1,890 @@
+
+One possible result of introducing `try` as a keyword be that the old `try!`
+macro would break. This could potentially be avoided but with great technical
+challenges.
+
+With the prospect of breaking [`try!`], a few notes are in order:
+
+1. `?` was stabilized in 1.13, November 2016, which is roughly 1.4 years since
+   the date this RFC was started.
+2. `try!` has been "deprecated" since then since:
+   > The `?` operator was added to replace `try!` and should be used instead.
+3. `try!(expr)` can in virtually all instances be automatically `rustfix`ed
+   automatically to `expr?`.
+4. There are very few questions on Stack Overflow that mention `try!`.
+5. ["The Rust Programming Language", 2nd edition](https://doc.rust-lang.org/book/second-edition/) (book) and "Rust by Example"
+   have both already removed all mentions of `try!`.
+
+> So overall I think it’s feasible to reduce the `try!` macro to a historical curiosity to the point it won’t be actively confusing to newbies coming to Rust.
+
+\- [kornel](https://internals.rust-lang.org/t/bikeshed-rename-catch-blocks-to-fallible-blocks/7121/49)
+
+However,
+
+1. There are still plenty of materials out there which mention `try!`.
+2. `try!` is essentially the inverse of `try { .. }`.
+
+> Purging from the “collective memories of Rustaceans and Rust materials” is not something that easy.
+
+\- [Manish Goregaokar](https://internals.rust-lang.org/t/bikeshed-rename-catch-blocks-to-fallible-blocks/7121/50)
+
+In the RFC author's opinion however, the sum total benefits of `try { .. }`
+seem to outweigh the drawbacks of the difficulty with purging [`try!`] from
+our collective memory.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Review considerations
+
+Among the considerations when picking a keyword are, ordered by importance:
+
+1. Fidelity to the construct's actual behavior.
+
+2. Precedent from existing languages
+    1. Popularity of the languages.
+    2. Fidelity to behavior in those languages.
+    3. Familiarity with respect to their analogous constructs.
+
+   See the [prior art][prior-art] the [rationale for try]
+   for more discussion on precedent.
+
+2. Brevity.
+
+[`Try`]: https://doc.rust-lang.org/nightly/std/ops/trait.Try.html
+
+4. Consistency with the naming of the trait used for `?` (the [`Try`] trait).
+
+5. Consistency with related standard library function conventions.
+
+6. Degree / Risk of breakage.
+
+7. Consistency with old learning material.
+
+    1. Inversely: The extent of the old learning material
+
+   That is, (in)consistency with `?` and the `try!()` macro.
+   If the first clause is called `try`,
+   then `try { }` and `try!()` would have essentially inverse meanings.
+
+## Rationale for `try`
+[rationale for try]: #rationale-for-try
+
+1. **Fidelity to the construct's actual behavior:** Very high
+2. **Precedent from existing languages:**
+    + `do`: Haskell, Idris
+    + `try`: A lot, see [prior-art]
```

Comment:
I don't understand what this is trying to tell me. Why is `do` mentioned here?

Reply:
Nothing ;) This is a copy error. Thank you for spotting it!